pkg/api/v1/pod/util_test.go

```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
 		}
 	}
 }
+
+func TestGetContainerStatus(t *testing.T) {
+	type ExpectedStruct struct {
+		status v1.ContainerStatus
+		exists bool
+	}
+
+	tests := []struct {
+		status   []v1.ContainerStatus
+		name     string
+		expected ExpectedStruct
```

Comment:
Add a short description for each test case. Include this in your `assert` call later to clearly show which test case this is.

```
    desc string
```

```
   assert.Equal(t, test.expected.status, resultStatus, t.desc)
``` 

Please also do this for all the test functions below.